src/utils/socketHooks.js

```diff
@@ -0,0 +1,49 @@
+import {
+  createElement,
+  createContext,
+  useState,
+  useEffect,
+  useContext
+} from 'react'
+import { Socket } from 'phoenix'
+import { useUserSessions } from '../pages/Account/SettingsSessions'
+import { useUser } from '../stores/user'
+
+export const HOST_NAME = new URL(process.env.REACT_APP_BACKEND_URL).hostname
+
+export const SocketContext = createContext({
+  socket: null,
+  showTabLimitModal: false,
+  setShowTabLimitModal: null
+})
+
+export const SocketProvider = ({ children }) => {
+  const { user } = useUser()
+  const [sessions] = useUserSessions()
+  const session = sessions.find(session => session.isCurrent)
+  const [socket, setSocket] = useState(null)
+  const [showTabLimitModal, setShowTabLimitModal] = useState(false)
+
+  useEffect(() => {
+    if (!user || !session || !session.jti) return
+    const _socket = new Socket(`wss://${HOST_NAME}/socket`, {
+      params: { jti: session.jti }
+    })
+    _socket.connect()
+    setSocket(_socket)
```

Comment:
Since `socket` from upper scope is not used in this closure, we can rename `_socket` to `socket`